provider/cloudflare/cloudflare.go

```diff
@@ -890,7 +928,11 @@ func groupByNameAndTypeWithCustomHostnames(records DNSRecordsMap, chs CustomHost
 		}
 		targets := make([]string, len(records))
 		for i, record := range records {
-			targets[i] = record.Content
+			if records[i].Type == "MX" {
+				targets[i] = fmt.Sprintf("%v %v", *record.Priority, record.Content)
```

Comment:
not covered with unit tests

![Screenshot 2025-06-22 at 10 10 10](https://github.com/user-attachments/assets/c59d27b2-3c6a-4e2b-be91-904b78ee2f35)